# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/CacheConstants.h
# Commit: 7b00a16ad21e
# Full Hash: 7b00a16ad21ee2b820b679199ae3806250d861e9
# Author: Nathan LaPre <nlapre@mozilla.com>
# Date: 2024-09-10 09:03:27
# Regressor Bug: 1794974
# File Overlap Count: 1
# Description:
#   Bug 1794974: Part 2: Add ability to query and send individual cache domains, r=Jamie
#   
#   This revision adds the capability of querying and sending information about
#   individual cache domains. It introduces the concept of active cache domains to
#   the accessibility service: cache domains that we think clients need. Conversely,
# ==============================================================================

diff -r 642b6d8c4d3b -r 7b00a16ad21e accessible/base/CacheConstants.h
--- a/accessible/base/CacheConstants.h	Mon Sep 09 23:02:19 2024 +0000
+++ b/accessible/base/CacheConstants.h	Mon Sep 09 23:02:19 2024 +0000
@@ -15,6 +15,7 @@
 
 class CacheDomain {
  public:
+  static constexpr uint64_t None = 0;
   static constexpr uint64_t NameAndDescription = ((uint64_t)0x1) << 0;
   static constexpr uint64_t Value = ((uint64_t)0x1) << 1;
   static constexpr uint64_t Bounds = ((uint64_t)0x1) << 2;
@@ -36,6 +37,7 @@
   // Used for MathML.
   static constexpr uint64_t InnerHTML = ((uint64_t)0x1) << 17;
 #endif
+  static constexpr uint64_t TextBounds = ((uint64_t)0x1) << 18;
   static constexpr uint64_t All = ~((uint64_t)0x0);
 };
 
@@ -235,10 +237,10 @@
   // AccAttributes, CacheDomain::Text
   // Text attributes; font, etc.
   static constexpr nsStaticAtom* TextAttributes = nsGkAtoms::style;
-  // nsTArray<int32_t, 4 * n>, CacheDomain::Text | CacheDomain::Bounds
+  // nsTArray<int32_t, 4 * n>, CacheDomain::TextBounds
   // The bounds of each character in a text leaf.
   static constexpr nsStaticAtom* TextBounds = nsGkAtoms::characterData;
-  // nsTArray<int32_t>, CacheDomain::Text | CacheDomain::Bounds
+  // nsTArray<int32_t>, CacheDomain::TextBounds
   // The text offsets where new lines start.
   static constexpr nsStaticAtom* TextLineStarts = nsGkAtoms::line;
   // nsString, CacheDomain::Value